app/controllers/api/blueprints_controller.rb

```diff
@@ -5,20 +5,15 @@ class BlueprintsController < BaseController
     before_action :set_additional_attributes, :only => [:show]
 
     def create_resource(_type, _id, data)
-      attributes = data.except("bundle")
-      blueprint = Blueprint.new(attributes)
-      bundle = data["bundle"]
-      create_bundle(blueprint, bundle) if bundle
-      blueprint.save!
-      blueprint
+      raise BadRequestError, 'Failed to create Blueprint - ui_properties cannot be empty' unless data['ui_properties']
+      validate_ui_properties(data['ui_properties'])
+      Blueprint.create!(data)
     end
 
     def edit_resource(type, id, data)
-      attributes = data.except("bundle")
+      validate_ui_properties(data['ui_properties']) if data['ui_properties']
```

Comment:
`ui_properties` is only needed to be valid at time of publishing. No need to validate here. It is not even required to exist at blueprint creation.